eth/tracers/api_test.go

```diff
@@ -544,6 +545,194 @@ func TestTraceTransactionWithRegistryDeployed(t *testing.T) {
+		},
+	}}
+
+	target := common.Hash{}
+	signer := types.HomesteadSigner{}
+	transferVal := big.NewInt(1000)
+	api := NewAPI(newTestBackend(t, 1, genesis, func(i int, b *core.BlockGen) {
+		// Transfer from account[0] to account[1]
+		//    value: 1000 wei
+		//    fee:   0 wei
+		tx, _ := types.SignTx(types.NewTransaction(uint64(i), accounts[1].addr, transferVal, params.TxGas, nil, nil, nil, nil, nil), signer, accounts[0].key)
+		b.AddTx(tx)
+		target = tx.Hash()
+	}))
+	tracerStr := "callTracer"
+	result, err := api.TraceTransaction(context.Background(), target, &TraceConfig{Tracer: &tracerStr})
+	if err != nil {
+		t.Errorf("Failed to trace transaction %v", err)
+	}
+
+	ret := new(callTrace)
+	if err := json.Unmarshal(result.(json.RawMessage), ret); err != nil {
+		t.Fatalf("failed to unmarshal trace result: %v", err)
+	}
+	expectedTrace := &callTrace{
+		Type:    "CALL",
+		From:    accounts[0].addr,
+		To:      accounts[1].addr,
+		Input:   hexutil.Bytes(common.Hex2Bytes("0x")),
+		Output:  hexutil.Bytes(common.Hex2Bytes("0x")),
+		Gas:     newRPCUint64(0),
+		GasUsed: newRPCUint64(0),
+		Value:   (*hexutil.Big)(transferVal),
+	}
+	if !jsonEqual(ret, expectedTrace) {
+		t.Fatalf("trace mismatch: \nhave %+v\nwant %+v", ret, expectedTrace)
+	}
+}
+
+// Use the callTracer to trace a CELO transfer made via the transfer
+// precompile (by sending this from the registered GoldToken contract).
+func TestCallTraceTransactionPrecompileTransfer(t *testing.T) {
+	// Invoke the transfer precompile by sending a transfer from an account
+	// registered as GoldToken in the mock registry
+	t.Parallel()
+	// Initialize test accounts
+	accounts := newAccounts(3)
+	goldToken := accounts[0]
+	registryProxyAddr := common.HexToAddress("0xce10")
+	registryImplAddr := common.HexToAddress("0xce11")
+	genesis := &core.Genesis{Alloc: core.GenesisAlloc{
+		goldToken.addr:   {Balance: big.NewInt(params.Ether)},
+		accounts[1].addr: {Balance: big.NewInt(params.Ether)},
+		accounts[2].addr: {Balance: big.NewInt(params.Ether)},
+		registryProxyAddr: { // Registry Proxy
+			Code: testutil.RegistryProxyOpcodes,
+			Storage: map[common.Hash]common.Hash{
+				common.HexToHash("0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc"): common.HexToHash("0xce11"),                                                             // Registry Implementation
+				common.HexToHash("0x91646b8507bf2e54d7c3de9155442ba111546b81af1cbdd1f68eeb6926b98d58"): common.HexToHash("0xd023"),                                                             // Governance Proxy
+				common.HexToHash("0x773cc8652456781771d48fb3d560d7ba3d6d1882654492b68beec583d2c590aa"): goldToken.addr.Hash(),                                                                  // GoldToken Implementation
+				common.HexToHash("0x2131a4338f6fb8d4507e234a7c72af8efefbbf2f1817ed570bce33eb6667feb9"): common.HexToHash("0x000000000000000000000000000000000000000000000000000000000000d007"), // Reserve Implementation
```

Comment:
looking at the lines above, it seems like you don't need all leading zeros.

Reply:
indeed!